VariationalBayes/ExponentialFamilies.py

```diff
@@ -116,6 +117,37 @@ def get_e_log_dirichlet(alpha):
     digamma_sum = sp.special.digamma(np.sum(alpha, 0, keepdims=True))
     return sp.special.digamma(alpha) - digamma_sum
 
+gh_loc, gh_weights = hermgauss(4)
+def get_e_logitnormal(lognorm_means, lognorm_infos, \
+            loc = gh_loc, weights = gh_weights):
+    # get the expectation of a logit normal distribution
+    e_sticks = np.array([np.dot(1/np.sqrt(np.pi) * weights,  \
+                sp.special.expit(np.sqrt(2) * loc * \
+                1/np.sqrt(lognorm_infos[k]) +\
+                lognorm_means[k])) \
+                for k in range(np.size(lognorm_means))])
+
+    return e_sticks
+
+def get_e_log_logitnormal(lognorm_means, lognorm_infos,\
+                            loc = gh_loc, weights = gh_weights):
+    # get expectation of Elog(X) and Elog(1-X), when X follows a logit normal
+
+    log_v = lambda x : np.log(sp.special.expit(x))
+    log_1mv = lambda x : np.log(1 - sp.special.expit(x))
+
+    e_log_v = np.array([np.dot(1/np.sqrt(np.pi) * weights,  \
+                log_v(np.sqrt(2) * loc * 1/np.sqrt(lognorm_infos[k]) +\
+                lognorm_means[k])) \
+                for k in range(np.size(lognorm_means))])
+
+    e_log_1mv = np.array([np.dot(1/np.sqrt(np.pi) * weights,  \
```

Comment:
Doing GH integration is always the same: you scale and shift the gh locations, evaluate the function at these points, and then take the sum scaled by the weights / sqrt(pi).

Please first write one function that does GH integration, without a for loop, for a generic function, which can be passed as an argument.  Then re-write ```get_e_logitnormal``` and ```get_e_log_logitnormal``` as special cases of that function.